src/commons/inference/client.py

```
@@ -0,0 +1,373 @@
+        logger.warning(
+            "Reached maximum iterations (%d) without final answer", max_iterations
+        )
+        raise InferenceIterationLimitError(
+            f"Maximum tool calling iterations ({max_iterations}) reached. "
+            "Please try again with a simpler query."
+        )
+
+    def _create_http_client(self, verify_ssl: bool, timeout: float) -> httpx.Client:
+        if not verify_ssl:
+            logger.warning("SSL verification disabled for %s", self.base_url)
+            return httpx.Client(verify=False, timeout=timeout)
```

Ruff (0.15.15)

[error] Probable use of `httpx` call with `verify=False` disabling SSL certificate checks

(S501)